test/integration/ci-operator-configresolver.sh

```diff
@@ -1,65 +1,88 @@
 #!/bin/bash
 source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
 
-rsync_tmp_dir=$(mktemp -d)
 function cleanup() {
-    rm -rf $rsync_tmp_dir
     os::test::junit::reconcile_output
     os::integration::configresolver::check_log
     os::cleanup::processes
 }
 trap "cleanup" EXIT
 
 suite_dir="${OS_ROOT}/test/integration/ci-operator-configresolver/"
-cp -a "${suite_dir}"/* "${BASETMPDIR}"
-cp -a "${OS_ROOT}/test/multistage-registry" "${BASETMPDIR}"
-actual="${BASETMPDIR}/actual"
-mkdir -p "${actual}"
-expected="${suite_dir}/expected"
+unused_dir=${suite_dir}/unused
 
 os::test::junit::declare_suite_start "integration/ci-operator-configresolver"
-# This test validates the ci-operator-configresolver tool
 
-os::integration::configresolver::start "${BASETMPDIR}/configs" "${BASETMPDIR}/multistage-registry/registry" "${BASETMPDIR}/config.yaml"
+test_validation() {
+    local out=${BASETMPDIR}/validation
+    local cmd=(
+        ci-operator-configresolver
+        --config "${unused_dir}/"
+        --jobs "${unused_dir}/"
+        --prow-config "${unused_dir}/unused.md"
+        --validate-only)
+    # Validate empty registry directory.
+    os::cmd::expect_success "${cmd[*]} --registry ${unused_dir}/"
+    # Unresolved parameters are valid at this stage.
+    os::cmd::expect_success \
+        "${cmd[*]} --registry ${suite_dir}/validation/unresolved/"
+    mkdir -p "${out}"
+    # Duplicate names at resolve time generate an error.
+    os::cmd::expect_code_and_text \
```

Comment:
This is crazy -- IMHO we do not gain much from this test and we should just omit the integration stuff entirely (it is checked at the unit level totally fine)

Reply:
I don't understand what is "crazy" about it.  It adds an actual bug report as an integration test.

We have found more than one bug exclusively via registry integration tests, and potentially have one right now (I haven't been able to track it yet).

Reply:
The crazy part is how complicated the test is

Reply:
Like IMHO testing the content of a specific error mesage is *so* much lower level than a full-blown run-the-binary integration test. What, specifically, does this test add that well-formed unit tests in Go could not?